File: EasySports-backen/src/main/java/com/easysports/model/Liga.java
package com.easysports.model;

import com.easysports.enums.Deporte;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Entidad que representa una Liga en la base de datos.
 * Mapea a la tabla "ligas".
 */
@Entity
@Table(name = "ligas")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Liga {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Deporte deporte;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "admin_id")
    private User admin;
}

File: EasySports-flutter/easy_sports_app/lib/src/models/liga.dart
class Liga {
  final int id;
  final String nombre;
  final String deporte;
  final int adminId;

  Liga({
    required this.id,
    required this.nombre,
    required this.deporte,
    required this.adminId,
  });

  factory Liga.fromJson(Map<String, dynamic> json) {
    return Liga(
      id: json['id'],
      nombre: json['nombre'],
      deporte: json['deporte'],
      adminId: json['adminId'],
    );
  }
}